wolfcrypt/test/test.c

```diff
@@ -10759,9 +10768,9 @@ WOLFSSL_TEST_SUBROUTINE int aesgcm_test(void)
 #endif
 #endif
 
-    byte resultT[sizeof(t1)];
-    byte resultP[sizeof(p) + AES_BLOCK_SIZE];
-    byte resultC[sizeof(p) + AES_BLOCK_SIZE];
+    ALIGN64 byte resultT[sizeof(t1)];
```

Comment:
Why is `ALIGN64` required?

Reply:
Shouldn't the alignment be handled in the AES GCM call? If not 64-bit aligned then allocate aligned buffer. If you want it in test.c for testing purposes it should not be for ALL cases, just for the Versal build. Make sense?

Reply:
Can I expect `XMALLOC()` to always return an at-least 64bit aligned buffer?

Reply:
Depends on RTOS, but generally no. You should allocate like this:

See this snippet in aes.c:1809 for AESNI as a reference:
```
        /* check alignment, decrypt doesn't need alignment */
        if ((wc_ptr_t)inBlock % AESNI_ALIGN) {
        #ifndef NO_WOLFSSL_ALLOC_ALIGN
            byte* tmp = (byte*)XMALLOC(AES_BLOCK_SIZE + AESNI_ALIGN, aes->heap,
                                                      DYNAMIC_TYPE_TMP_BUFFER);
            byte* tmp_align;
            if (tmp == NULL)
                return MEMORY_E;

            tmp_align = tmp + (AESNI_ALIGN - ((wc_ptr_t)tmp % AESNI_ALIGN));

            XMEMCPY(tmp_align, inBlock, AES_BLOCK_SIZE);
            AES_ECB_encrypt(tmp_align, tmp_align, AES_BLOCK_SIZE,
                    (byte*)aes->key, aes->rounds);
            XMEMCPY(outBlock, tmp_align, AES_BLOCK_SIZE);
            XFREE(tmp, aes->heap, DYNAMIC_TYPE_TMP_BUFFER);
            return 0;
        #else
            WOLFSSL_MSG("AES-ECB encrypt with bad alignment");
            WOLFSSL_ERROR_VERBOSE(BAD_ALIGN_E);
            return BAD_ALIGN_E;
        #endif
        }
```